packages/web-pkg/src/components/SideBar/FileSideBar.vue

```diff
@@ -334,6 +346,42 @@ export default defineComponent({
       sharesStore.setLoading(false)
     }).restartable()
 
+    watch(
+      () => [...unref(panelContext).items, props.isOpen],
+      async () => {
+        if (unref(panelContext).items?.length !== 1) {
+          // don't load additional metadata for empty or multi-select contexts
+          return
+        }
+
+        const resource = unref(panelContext).items[0]
+
+        if (loadVersionsTask.isRunning) {
+          loadVersionsTask.cancelAll()
+        }
+
+        if (
+          !resource.isFolder &&
+          !isProjectSpaceResource(resource) &&
+          !isIncomingShareResource(resource) &&
+          !unref(isPublicFilesLocation) &&
+          !unref(isTrashLocation) &&
+          !unref(isSharedWithOthersLocation) &&
+          !unref(isSharedViaLinkLocation)
+        ) {
+          try {
+            isVersionsLoading.value = true
```

Comment:
Instead of maintaining a `isVersionsLoading` ref manually, you should have a `computed` in place which just returns `loadVersionsTask.isRunning`. Less error-prone and built-in into the `vue-concurrency` lib.